main/guides/getting-started/contract-rpc.md

```diff
@@ -0,0 +1,410 @@
+# Smart Wallet Dapp Architecture
+
+The [Agoric Platform](../platform/) consists of smart contracts
+and services such as [Zoe](../zoe/) running in a [Hardened JavaScript](../js-programming/hardened-js.md) VM running on top of a Cosmos SDK consensus layer. Clients interact with the consensus layer by making
+queries and submitting messages in signed transactions. In the Smart
+Wallet Architecture, dapps consist of
+
+- Hardened JavaScript smart contracts
+- clients that can submit offers and query status via the consensus layer
+
+![smart wallet dapp sequence diagram](./assets/sw-dapp-arch.svg)
+
+1. A client formats an offer, signs it, and broadcasts it.
+2. The offer is routed to the `walletFactory` contract, which finds (or creates) the `smartWallet` object associated with the signer's addres and uses it to execute the offer.
+3. The `smartWallet` calls `E(zoe).offer(...)` and monitors the status of
+   the offer, emitting it for clients to query.
+4. Zoe escrows the payments and forwards the proposal to the contract
+   indicated by the offer.
+5. The contract tells Zoe how to reallocate assets.
+6. Zoe ensures that the reallocations respect offer safety and then provides payouts accordingly.
+7. The client's query tells it that the payouts are available.
+
+## Signing and Broadcasting Offers
+
+One way to sign and broadcast offers is with the `agd tx ...` command. For example:
+
+```sh
+agd tx swingset wallet-action --allow-spend "$ACTION" \
+ --chain-id=agoriclocal --from=acct1
+```
+
+<!-- TODO: discuss agoric wallet send? -->
+
+Another is using a wallet signing UI such as Keplr via the [Keplr API](https://docs.keplr.app/api/).
```

Comment:
A screenshot of signing with Keplr would be nice.

I looked high and low for one.
  - wallet UI docs? nope
  - docs.inter.trade? nope
  - keplr docs? nope.

Reply:
a TODO comment  for a screenshot may be worthwhile